api/src/main/java/com/epam/pipeline/manager/cloud/CloudFacadeImpl.java

```diff
@@ -173,6 +174,20 @@ public boolean instanceExists(final Long regionId, final String instanceId) {
         return getInstanceService(region).getInstanceTerminationState(region, instanceId);
     }
 
+    @Override
+    public List<InstanceType> getAllInstanceTypes(final Long regionId, final Boolean spot) {
+        if (regionId == null) {
+            return (List<InstanceType>) instancePriceServices.values()
+                    .stream()
+                    .map(priceService -> priceService.getAllInstanceTypes(regionId, spot))
```

Comment:
I'd suggest to split this into two methods, since they implement different logic.